src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -241,6 +243,25 @@ int sd_dhcp_client_id_to_string(const void *data, size_t len, char **ret) {
         return 0;
 }
 
+int sd_dhcp_client_set_state_callback(
```

Comment:
Then, please make also this private, that is, drop `sd_` prefix and declare it in dhcp-internal.h or so.

Reply:
I put these internal interfaces that are intended to be used by the dhcp client within the `dhcp-client-internal.h` and `dhcp6-client-internal.h` headers. Let me know if that isn't the right approach.